### PR TITLE
Changes to add DPC++ testing in Github CI

### PR DESCRIPTION
**Pull request by KumudhaN**
Changes to add DPC++ testing in Github CI

### CODE REVIEW

File: .scripts/build_dpcpp.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+set -ev
+
+###########################
+# Get DPCPP
+###########################
+wget --no-verbose https://github.com/intel/llvm/releases/download/sycl-nightly/20221201/dpcpp-compiler.tar.gz -O dpcpp-compiler.tar.gz
+rm -rf /tmp/dpcpp-latest && mkdir /tmp/dpcpp-latest/
+tar -xzf dpcpp-compiler.tar.gz -C /tmp/dpcpp-latest --strip-components 1
+ls -R /tmp/dpcpp-latest/
```
Comment (hjabird) [inline .scripts/build_dpcpp.sh:11]: Newline

File: .scripts/build_dpcpp.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -ev
+
+###########################
+# Get DPCPP
+###########################
+wget --no-verbose https://github.com/intel/llvm/releases/download/sycl-nightly/20221201/dpcpp-compiler.tar.gz -O dpcpp-compiler.tar.gz
```
Comment (hjabird) [inline .scripts/build_dpcpp.sh:8]: `dpcpp-latest` suggests that this automatically updates to the latest version.

File: Dockerfile [hunk 3]
```diff
@@ -57,15 +57,15 @@ ENV TARGET=${target}
...
         /tmp/ComputeCpp-latest/bin/computecpp_info && \
-        COMPUTECPP_TARGET="intel:cpu" ./build.sh /tmp/ComputeCpp-latest /tmp/OpenBLAS/build; \
+        COMPUTECPP_TARGET="intel:cpu" ./build.sh --compiler computecpp --computecppdir /tmp/ComputeCpp-latest --openblas /tmp/OpenBLAS/build; \
       fi \
+    elif [ "${SYCL_IMPL}" = 'DPCPP' ]; then \
+      export LD_LIBRARY_PATH="/tmp/dpcpp-latest/lib" && ./build.sh --compiler dpcpp --openblas /tmp/OpenBLAS/build; \
     else \
       echo "Unknown SYCL implementation ${SYCL_IMPL}"; return 1; \
-    fi
+    fi
```
Comment (hjabird) [inline Dockerfile:71]: Newline?

**Review by Rbiessy — APPROVED** [review]
I'm not a fan of this `build.sh` script, I think it's just adding a layer of indirection and renaming the CMake flags to bash flags but I won't block on this.

**Review by KumudhaN — COMMENTED** [review]
Removed the build.sh script

**Review by muhammad-tanvir-1211 — APPROVED** [review]

**Comment by Rbiessy** [comment]
LGTM, thanks!

**Pull request merged**